### DISCUSSION TITLE
Older Gotek faster in Akai S950 than newer Gotek

### DISCUSSION THREAD

**Discussion by rCket** [keirf/flashfloppy#816] [General]
Hi.
I have an S950 and both an SFRC922D, ARM STM32F105 Gotek drive and an SFRKC30.AT4.35, ARTERY AT32F435 Gotek drive.

I pulled out the older Gotek and fitted thenew one with the larger OLED.
Now, after switching back and forth between them the newer Gotek is nearly 100% slower when loading a disk.

Tested with one disk image that almost fills the S950 RAM, it loads with the older Gotek in 1 minute 14 seconds and with the newer Gotek at 2 minutes 20 seconds.

I've been through the FF.CFG file many times but nothing alters the fact that it's slower.

Any ideas?

Thanks

**Comment by rCket** [comment]
I remember, years ago, that the HxC floppy emulator needed a certain "resistor mod" for it to work properly in an S950. I wonder if that has anything to do with the slowness? My older Gotek was bought from somebody who was selling it as S950 compatible. I've looked but can't see any resistor mods on it.

Also, FF.CFG:
`
host=akai
interface=akai-s950
'

I also tried interface=jppc-hdout as I see that "akai-s950" is now depreciated. Same effect.

**Reply by keirf** [reply]
Same firmware version on both Goteks? Same usb stick? Reset any cached config by holding both buttons (or rotary encoder) for a few seconds with no stick inserted until the display changes.

**Reply by rCket** [reply]
3.41. Yeah, I always do a factory reset.
Same usb stick. 

Also, I have two S950’s atm so have also switched Goteks between both machines to rule out a problem with the S950 itself. So, after trying both Goteks in both machines, the older Gotek is nearly twice as fast in both S950’s.

**Reply by keirf** [reply]
Have you tried more than one usb stick? It is indeed hard to understand how same firmware version, same stick, same config would have a 2x performance difference.

**Reply by keirf** [reply]
If you can supply an offending image file I might find some time to try it out in some simple ways and look for performance differences.

**Reply by rCket** [reply]
Sent file and link to a few vids showing it.

**Comment by rCket** [comment]
Sorted.

'
side-select-glitch-filter = 100
'

Made it load in 1 minute. Faster than the older Gotek.

**Reply by keirf** [reply]
It must be a very short glitch if the old stm32f105 misses it. Probably even a value of say 5 would be more than sufficient rather than 100?

**Comment by rCket** [comment]
> It must be a very short glitch if the old stm32f105 misses it. Probably even a value of say 5 would be more than sufficient rather than 100?

Doesn't seem to be the case.

5 takes 2:15.
50 takes 1:32.
75 takes 1:07.
100 and above takes 1:02.

**Comment by keirf** [comment]
Thanks for the testing. That's very interesting. I guess there are glitches of various lengths up to 100us. But then I don't understand why AT32F435 is disproportionately affected. Perhaps the MCU is fast enough to get committed to USB I/O on the "wrong" side of the disk and remains blocked a while even after the head signal settles. That's a bit of a guess though. At 1us+ the old 105 chip would see these glitches too, and at 10us+ I would expect it to be getting bound up in the same I/O paths. Perhaps the speed difference is more than I think.
